instrumentation/grpc/RATIONALE.md

```diff
@@ -25,3 +25,14 @@ record the application exception.
 In short, we choose to not mask the gRPC status with a potential application
 exception on close. In worst case, if there is no instrumentation for the layer
 that throws, its trace and span ID could be known via log correlation.
+
+## Why don't we use RpcClientHandler or RpcServerHandler for unexpected errors?
```

Comment:
@anuraaga since we now have a more compelling story about parsing, I agree that we should keep it nicer on the happy path vs the edge cases. This follows up your points on the other pull request I hope. Otherwise, let me know how to change it.